### ISSUE TITLE
Even if stock management is disabled, show "on backorder" text in the product catalog if set

### ISSUE THREAD

**Issue by aj-adl** [woocommerce/woocommerce#20152]
**Describe the bug**

Products that are "On backorder" do not communicate this to the customer when the global setting at WooCommerce > Products > Inventory "Enable Stock Management" is set to `false` (unchecked)

![image](https://user-images.githubusercontent.com/3295954/40297166-01986ff0-5d1e-11e8-95ee-568414330c3d.png)

**To Reproduce**

1. Go to`WooCommerce > Products > Inventory via wp-admin` or `wp option update "woocommerce_manage_stock" "no"` via CLI

2. Create a Product and set it's stock status to 'On Backorder'
![image](https://user-images.githubusercontent.com/3295954/40297330-970d54b0-5d1e-11e8-8948-4be8945bda43.png)

4. Browse site, add product to the cart, try and checkout etc

Issue 1: No changes to product metabox
- cannot select whether to allow backorders or not (just defaults to yes)
- cannot select whether to notify the customer that a product is on backorder

Issue 2: As a result of the behaviour above, a product with 'On Backorder' does not notify the customer 
![image](https://user-images.githubusercontent.com/3295954/40297357-b41d99ca-5d1e-11e8-8086-26b9eadacc5b.png)

**Expected behavior**
Expected the behaviour to be more consistent with the backorder options when stock management is turned on, ie: have the option to notify customers, or disallow purchasing of backordered products (in case the business is just using these as an 'internal' status)

Here are some screenshots of the expected behaviour - in this example I an using the same test store as above, but with `woocommerce_manage_stock` set to to true

Product Meta Box
![image](https://user-images.githubusercontent.com/3295954/40297757-ff51c096-5d1f-11e8-9d34-097ff8f36b4b.png)

Result on the front end (when "allow but notify customer" is selected, by far the preferred option from clients"
![image](https://user-images.githubusercontent.com/3295954/40297814-312252d4-5d20-11e8-8470-d48733d903bc.png)

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
### WordPress Environment ###

Home URL: http://testbed.test
Site URL: http://testbed.test/wordpress
WC Version: 3.3.5
Log Directory Writable: ✔
WP Version: 4.9.6
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_AU

### Server Environment ###

Server Info: nginx/1.13.10
PHP Version: 7.2.4
PHP Post Max Size: 128 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.54.0
SecureTransport

SUHOSIN Installed: –
MySQL Version: 5.7.21
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.5
WC Database Prefix: frame_
Total Database Size: 1.79MB
Database Data Size: 0.95MB
Database Index Size: 0.84MB
frame_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB
frame_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
frame_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
frame_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
frame_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
frame_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
frame_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
frame_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
frame_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
frame_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
frame_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
frame_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
frame_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
frame_woocommerce_log: Data: 0.02MB + Index: 0.02MB
frame_commentmeta: Data: 0.02MB + Index: 0.03MB
frame_comments: Data: 0.02MB + Index: 0.08MB
frame_links: Data: 0.02MB + Index: 0.02MB
frame_options: Data: 0.41MB + Index: 0.02MB
frame_postmeta: Data: 0.02MB + Index: 0.03MB
frame_posts: Data: 0.02MB + Index: 0.06MB
frame_termmeta: Data: 0.02MB + Index: 0.03MB
frame_terms: Data: 0.02MB + Index: 0.03MB
frame_term_relationships: Data: 0.02MB + Index: 0.02MB
frame_term_taxonomy: Data: 0.02MB + Index: 0.03MB
frame_usermeta: Data: 0.02MB + Index: 0.03MB
frame_users: Data: 0.02MB + Index: 0.05MB
frame_wc_download_log: Data: 0.02MB + Index: 0.03MB
frame_wc_webhooks: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

page: 6
post: 2
product: 1

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (3) ###

WooCommerce PayPal Express Checkout Gateway: by WooCommerce – 1.5.4
WooCommerce Stripe Gateway: by WooCommerce – 4.1.3
WooCommerce: by Automattic – 3.3.5

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.3.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
```
</details>

**Comment by mikejolley** [comment]
> Expected the behaviour to be more consistent with the backorder options when stock management is turned on, ie: have the option to notify customers, or disallow purchasing of backordered products (in case the business is just using these as an 'internal' status)

"internal status" is not a use case we cover. If it's internal, manage it outside of WC.

Having these options doesn't make much sense if you're managing stock outside of WC. If you disable stock management, you may expect these settings to be hidden to avoid extra noise. I disagree they should be shown.

However, it might make sense to show 'on backorder' if manually set, like it does for 'out of stock'. I'll update the issue title and mark as a feature request.

Thanks

**Comment by woocommercebot** [comment]
Thanks for the suggestion @aj-adl.

We'll keep an eye on the [popularity of this request](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#feature-requests) :)

**Comment by LiamKarlMitchell** [comment]
Is it easily possible to make an extension or function hook to put in my theme to always show the on backorder notification if the product is on backorder, regardless of the stock managing status?

Advice on the right direction to go in implementing this would be appreciated.

**Comment by ynthrepic** [comment]
Chiming in after having searched for this feature. The back order feature is useful as a flag when such products can be acquired despite being technically out of stock. Out of stock prevents ordering, after all, and is more useful for hiding a temporarily unavailable item.

I think of 'Backorder' as a kind of 'Order anyway' option. This just seems like a logical feature to have, even if comprehensive stock management is disabled.

**Comment by rwkyyy** [comment]
I had to do some custom coding to get something near this, funny how something so tiny can make you lose almost 4 hours of your work time.

My client wanted this feature because it's stock was not maintained in the website and having a big network of stores is already a pain to maintain, syncing with the website would even cost more labor.

I ended overwriting the availability message with a custom one (input based) for the client so it can add whatever he wants and as long as the product is available then it can be ordered.
